internal/prober/prober.go

```diff
@@ -176,3 +190,17 @@ func (p *Prober) doProbe(client kubernetes.Interface) error {
 	}
 	return nil
 }
+
+func (p *Prober) setBackOffIfThrottlingError(err error) {
+	if apierrors.IsTooManyRequests(err) {
+		p.l.V(4).Info("API server is throttled, backing off", "backOffDuration", backOffDurationForThrottledRequests.Seconds())
+		p.resetBackoff(backOffDurationForThrottledRequests)
```

Comment:
the backoff handling doesn't appear correct, but since you have refactored this and new logic seems OK, will wait for next commit.